pkg/api/handlers/compat/containers_archive.go

```diff
@@ -1,12 +1,377 @@
+		return
+	}
+
+	if query.Path == "" {
+		utils.Error(w, "Bad Request.", http.StatusBadRequest, errors.New("missing `path` parameter"))
+		return
+	}
+
+	containerName := utils.GetName(r)
+
+	ctr, err := runtime.LookupContainer(containerName)
+	if errors.Cause(err) == define.ErrNoSuchCtr {
+		utils.Error(w, "Not found.", http.StatusNotFound, errors.Wrap(err, "the container doesn't exists"))
+		return
+	} else if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, err)
+		return
+	}
+
+	mountPoint, err := ctr.Mount()
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to mount the container"))
+		return
+	}
+
+	defer func() {
+		if err := ctr.Unmount(true); err != nil {
+			logrus.Warnf("failed to unmount container %s: %q", containerName, err)
+		}
+	}()
+
+	opts := copier.StatOptions{}
+
+	mountPoint, path, err := fixUpMountPointAndPath(runtime, ctr, mountPoint, query.Path)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, err)
+		return
+	}
+
+	stats, err := copier.Stat(mountPoint, "", opts, []string{filepath.Join(mountPoint, path)})
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to get stats about file"))
+		return
+	}
+
+	if len(stats) <= 0 || len(stats[0].Globbed) <= 0 {
+		errs := make([]string, 0, len(stats))
+
+		for _, stat := range stats {
+			if stat.Error != "" {
+				errs = append(errs, stat.Error)
+			}
+		}
+
+		utils.Error(w, "Not found.", http.StatusNotFound, fmt.Errorf("file doesn't exist (errs: %q)", strings.Join(errs, ";")))
+
+		return
+	}
+
+	statHeader, err := statsToHeader(stats[0].Results[stats[0].Globbed[0]])
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, err)
+		return
+	}
+
+	w.Header().Add("X-Docker-Container-Path-Stat", statHeader)
+
+	if r.Method == http.MethodGet {
+		idMappingOpts, err := ctr.IDMappings()
+		if err != nil {
+			utils.Error(w, "Not found.", http.StatusInternalServerError,
+				errors.Wrapf(err, "error getting IDMappingOptions"))
+			return
+		}
+
+		destOwner := idtools.IDPair{UID: os.Getuid(), GID: os.Getgid()}
+
+		opts := copier.GetOptions{
+			UIDMap:             idMappingOpts.UIDMap,
+			GIDMap:             idMappingOpts.GIDMap,
+			ChownDirs:          &destOwner,
+			ChownFiles:         &destOwner,
+			KeepDirectoryNames: true,
+		}
+
+		w.WriteHeader(http.StatusOK)
+
+		err = copier.Get(mountPoint, "", opts, []string{filepath.Join(mountPoint, path)}, w)
+		if err != nil {
+			logrus.Error(errors.Wrapf(err, "failed to copy from the %s container path %s", containerName, query.Path))
+			return
+		}
+	} else {
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func handlePut(w http.ResponseWriter, r *http.Request, decoder *schema.Decoder, runtime *libpod.Runtime) {
+	query := struct {
+		Path string `schema:"path"`
+		// TODO handle params below
```

Comment:
I am cool to merge this PR with the TODOs open. I can pick these work items up once merged.

Reply:
Great. I don't know how I would fix that quickly. `buildah/copier` doesn't have the option for `NoOverwriteDirNonDir`. But `github.com/containers/storage/pkg/chrootarchive` `Untar()` has the option I could use that instead of `buildah/copier`.